pkg/controller/job/job_controller.go

```diff
@@ -974,6 +974,27 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 	return manageJobErr
 }
 
+func (jm *Controller) newFailureCondition(reason, message string) *batch.JobCondition {
```

Comment:
This could work - it was my original idea, but then I thought that the functions `newFailureCondition` and `newSuccessCondition` have another benefit - they avoid duplication for setting the `ConditionTrue`, and `jm.clock.Now()`. So, maybe they have a right to exist even when the feature gates graduate. WDYT?

Reply:
I would prefer less indirection after the graduation and duplicate the `ConditionTrue`, and `jm.clock.Now()`, but I suppose it is a matter of personal preference.